### PR TITLE
HK13 - Restrict nether portals in safe zone only

### PR DESCRIPTION
**Pull request by DmitryRendov**
Address #13

Test cases:
* Check that players CAN'T create portals in safe zone overworld
* AND in the same area with scale factor /8 in the Nether world
* Only humans can light portals (no dispensers or fireballs)

### CODE REVIEW

**Comment by DmitryRendov** [comment]
Testing has showcased that I broke commands blocking for some reason)

File: src/main/java/me/drendov/HouseKeeping/Listeners.java [hunk 1]
```diff
@@ -75,22 +79,51 @@ public void onPlayerToggleFlight(PlayerToggleFlightEvent event) {
...
+        if (player.isOp() || player.hasPermission("housekeeping.bypass") || player.hasPermission("housekeeping.bypass.portal"))
+            return;
+
+        HashMap getSafeZoneArea = HouseKeeping.getInstance().config.getSafeZoneArea(player);
+        Location loc1 = (Location) getSafeZoneArea.get("loc1");
+        Location loc2 = (Location) getSafeZoneArea.get("loc2");
+        if (this.isInRect(player, loc1, loc2)) {
+            if (event.getReason() == PortalCreateEvent.CreateReason.FIRE) {
+                if (HouseKeeping.getInstance().config.getRulePreventPortalCreation()) {
+                    HouseKeeping.sendMessage(player, TextMode.Err, Messages.DenyNetherPortalMsg);
```
Comment (AlexeyFilich) [inline src/main/java/me/drendov/HouseKeeping/Listeners.java:118]: Should check not if player is in safe zone, but if portal (or block player tries to put on fire) is in safe zone. With current player-based check you can light a portal inside the safe zone by placing yourself outside. (line 115)

**Review by AlexeyFilich — CHANGES_REQUESTED** [review]
Command blocking is working fine, you probably forgot you changed safe zone position from 128 to 1024.

Other testing with portals did not showed any bugs.

File: src/main/java/me/drendov/HouseKeeping/Listeners.java (hunk 1 above)
Comment (DmitryRendov) [reply src/main/java/me/drendov/HouseKeeping/Listeners.java:118]: I'm aware of it. It's hard to do as for me)
having big safe zone (thousands of blocks), 2-3 non-safe blocks are OK unless this plugin will be published and we have to fix that.
raised #15 to address this issue later.

**Review by AlexeyFilich — APPROVED** [review]

**Pull request merged**